02-esc/modal.js

```diff
@@ -76,13 +81,30 @@ async function checkAvailableTimes(selectedDate, challengeId) {
 // -- MODAL STEP 2 --
 
 function showAvailableTimes(slot) {
-	slot = slot;
 	let option = document.createElement("option");
 	option.innerHTML = slot;
 	let select = document.querySelector("#time");
 	select.appendChild(option);
 }
 
+function possibleNumberOfParticipants(min, max) {
+    let numberOfOptions = max - min;
+    
+    for(i=0; i <= numberOfOptions; i++) {
+        createOptionForParticipants()
+        min++;    
+    }
+
+    function createOptionForParticipants() {
+        let participantsOption = document.createElement("option");
+        participantsOption.innerHTML = `${min} participants`;
+        let participantsSelect = document.querySelector("#participants");
+        participantsSelect.appendChild(participantsOption);
+
+    } 
+
+}
+
```

Comment:
The scoping in this function is not great, as it relies on outside data instead of passing the necessary data around in a way that can be easily followed. Here are my concerns:

* The for loop calls a function (`createOptionForParticipants()`) without passing any data, so it looks to me when I'm reading the for loop (without reading the details of the function) like it would do the same thing over and over again, which seems odd
* The `min` variable is incremented in the for loop, but I can't see a reason why unless I go and read the function.
* The `createOptionForParticipants()` function uses the `min` variable from it's parent scope, which is easily missed when glancing over it. If it had been a parameter instead, it would be so much more clear to a reader of the code what is going on.

I would change the for loop to look like this (and make the corresponding changes to `createOptionForParticipants()`):

```js
for (let i = 0; i < numberOfOptions; i++) {
  createOptionForParticipants(min + i);
}
```